python/tk_multi_workfiles/work_files.py

```diff
@@ -754,6 +754,23 @@ def select_work_area(self, mode=SelectWorkAreaForm.SELECT_WORK_AREA):
 
             do_new = widget.do_new_scene
 
+            # widget.context contains the selected context (shot or entity)
+            if os.getenv('REZ_USED_REQUEST'):
+                # we are under Rez, some entity may not be compatible with current one..
+                # delayed import on purpose, not available outside of rez:
+                new_ctx = widget.context
+                from rdo_rez_launcher.tools import start_app_if_new_entity_incompatible
+                if start_app_if_new_entity_incompatible(
+                    lambda title, message: QtGui.QMessageBox.Yes == QtGui.QMessageBox.question(
+                        None, title, message, QtGui.QMessageBox.Yes | QtGui.QMessageBox.No),
+                    new_ctx.tank.shotgun, new_ctx.entity['name'],
+                    tank_config_name=new_ctx.tank.configuration_name
+                ):
+                    # this means the new entity isn't compatible with current one and we stop here,
```

Comment:
Had to ask you what you meant by "compatible". Apparently what you mean is that you check for rez package conflicts since there could be different packages that can clash between different entities.

That makes sense but I think this line could be a bit more explanatory. ;)